backend/question-service/index.js

```diff
@@ -27,17 +27,33 @@ const PORT = 8002;
 
 const db = require("./models");
 const dbConfig = require("./config/db.config.js");
+const { MongoMemoryServer } = require('mongodb-memory-server');
 
-db.mongoose
+if (process.env.JASMINE) {
```

Comment:
Should this be in the implementation? It bears no relevance outside of testing, my opinion is that it should be elsewhere

Reply:
Since the unit tests use the index.js file, its needed to determine if its running in a test instance or not. If it is then it runs the database as a mock using the `mongodb-memory-server`.

Reply:
On principle anything related to test code, especially something explicit like this, should not be present in any code for deployment. I would favour the removal of this and see if there's a manner in which you can set up a mock to listen for a connection via another set of configs set through an environment file.

Reply:
Can I check what requests are being tested with the stub

Reply:
The stub is basically a mock of the database since these tests are to be run outside the container, all tests involving any controller with a database is run with the stub.

As for the environmental variables, would it be better to have a "testing copy" of the index.js to serve the same purpose? Considering all the index.js files are setup in the same manner?